### PR TITLE
[WFCORE-4239] WARN if system-property is already set and is being overridden

### PR DESCRIPTION
**Pull request by soul2zimate**
https://issues.jboss.org/browse/WFCORE-4239

### CODE REVIEW

File: server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java [hunk 1]
```diff
@@ -130,6 +133,18 @@ void parseSystemProperties(final XMLExtendedStreamReader reader, final ModelNode
             if (!setName) {
                 throw ParseUtils.missingRequired(reader, Collections.singleton(NAME));
             }
+
+            Properties properties = WildFlySecurityManager.getSystemPropertiesPrivileged();
```
Comment (jmesnil) [inline server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java]: you could read the system properties outside the `while` block

File: server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java [hunk 2]
```diff
@@ -130,6 +133,18 @@ void parseSystemProperties(final XMLExtendedStreamReader reader, final ModelNode
...
+            try {
+                String newPropertyValue = SystemPropertyResourceDefinition.VALUE.resolveValue(ExpressionResolver.SIMPLE, op.get(VALUE)).asString();
+                String oldPropertyValue = properties.getProperty(name);
+                if (oldPropertyValue != null && !oldPropertyValue.equals(newPropertyValue)) {
+                    ControllerLogger.ROOT_LOGGER.systemPropertyAlreadyExist(name, oldPropertyValue, newPropertyValue);
+                }
+            } catch (OperationFailedException e) {
+                ServerLogger.AS_ROOT_LOGGER.tracef(e, "Failed to set property %s at parse time, it will be set later in the boot process", name);
```
Comment (jmesnil) [inline server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java]: is the message correct? You are not setting the property here, you just check its resolved value against the `System.getProperties()`.

Comment (soul2zimate) [reply server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java]: Thanks. This message is copied from trace message a few lines below, I will adjust it.

File: server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java (hunk 1 above)
Comment (soul2zimate) [reply server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java]: I have moved it outside.

**Pull request merged**